### PR TITLE
Remove appveyor.yml from V2

### LINKED ISSUE #2898: Remove appveyor.yml and correct typos in the build pipeline from V2, V3, V4
Currently, the `dev`, `v3.x`, and `v4.x` branches contain `appveyor.yml`, which is the outdated build pipeline for Core Tools V1, originally hosted on AppVeyor. We have moved the pipeline to Azure DevOps, and as such `appveyor.yml` (and all references to AppVeyor in `azure-pipelines.yml`) should be removed from these branches.

### PR DESCRIPTION
**Pull request by michaelpeng36**
### Issue describing the changes in this PR

Resolves #2898 for Core Tools V2. Added some minor changes to correct exception messages and change `Invoke-Expression` to `& {}`

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -1,24 +1,24 @@
-if ($env:APPVEYOR_REPO_BRANCH -eq "disabled") {
+if ($env:DEVOPS_REPO_BRANCH -eq "disabled") {
     Set-Location ".\src\Azure.Functions.Cli"
-    $result = Invoke-Expression -Command "NuGet list Microsoft.Azure.Functions.JavaWorker -Source  https://ci.appveyor.com/NuGet/azure-functions-java-worker-fejnnsvmrkqg -PreRelease"
+    $result = & { "NuGet list Microsoft.Azure.Functions.JavaWorker -Source  https://ci.appveyor.com/NuGet/azure-functions-java-worker-fejnnsvmrkqg -PreRelease" }
     $javaWorkerVersion = $result.Split()[1]
     Write-host "Adding Microsoft.Azure.Functions.JavaWorker $javaWorkerVersion to project" -ForegroundColor Green
-    Invoke-Expression -Command "dotnet add package Microsoft.Azure.Functions.JavaWorker -v $javaWorkerVersion -s  https://ci.appveyor.com/NuGet/azure-functions-java-worker-fejnnsvmrkqg"
+    & { "dotnet add package Microsoft.Azure.Functions.JavaWorker -v $javaWorkerVersion -s  https://ci.appveyor.com/NuGet/azure-functions-java-worker-fejnnsvmrkqg" }
```
Comment (Francisco-Gamino) [inline build.ps1]: Thank you for updating these.

File: build.ps1 [hunk 2]
```diff
@@ -32,7 +32,7 @@ $simulateReleaseBuild = $null
 if (-not([bool]::TryParse($env:IsReleaseBuild, [ref] $isReleaseBuild) -and
```
Comment (Francisco-Gamino) [inline build.ps1:32]: Something to consider:
To set a release build, you could do the following: 1) Set a variable in the pipeline 2) Or analyze the branch from which the build is being executed. 

Is there a reason why we need both of these variable `$env:SimulateReleaseBuild` and `$env:IsReleaseBuild`?

Comment (michaelpeng36) [reply build.ps1:32]: Yes, `$env:IsReleaseBuild` is evaluated at runtime to determine if the branch is a release branch, i.e. the branch name is of the form `release_2.0*`. `$env:SimulateReleaseBuild` is a variable to be overridden when the build will not be used for a release, but we want to check that the build process is working properly for release builds (specifically, to check release-specific build steps like SBOM generation or signing). What do you think? Should we combine these in some way?

**Review by Francisco-Gamino — APPROVED** [review]
LGTM

**Pull request merged**